src/library/scala/collection/ArrayOps.scala

```diff
@@ -1637,4 +1637,12 @@ final class ArrayOps[A](private val xs: Array[A]) extends AnyVal {
     *  @return `true` if this array has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = mutable.ArraySeq.make(xs).endsWith(that)
+
+  /** Ensure we are providing an array of the correct element type.
+   */
+  private def narrowly(ys: Array[_]): Array[A] =
+    if (implicitly[ClassTag[A]].runtimeClass.isAssignableFrom(ys.getClass.getComponentType))
```

Comment:
```suggestion
    if (xs.getClass.getComponentType == ys.getClass.getComponentType))
```

We should not go into subtyping here, returning an `Array[String]` when `A == Object` could lead to `ArrayStoreException`s.

Reply:
I assumed the Y is wider than A, but I also did not devise a use case where Y is narrower than A. I took this suggestion before just `toArray`.